client-c/main.cpp

```diff
@@ -261,7 +261,7 @@ void runKServer(httplib::Server *svr) {
   block* final_config = take_steps(K_DEPTH, init_config);
   if (FLAGS_dump) printConfiguration("/dev/stderr", final_config);
   close(K_WRITE_FD);
-  close(K_READ_FD);
+  close(output[1]);
```

Comment:
So we're only closing `output[1]` and `input[1]`, why not also `output[0]` and `input[0]`?

Reply:
Closing `output[0]` could interrupt the http server while it is trying to `read` a final rpc message from the K server. `input[0]` is the file descriptor which the (now closed) K server reads from. AFAICT there's no specification that pipes need to be fully closed before the process ends, so I'm just leaving it alone.